core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -653,15 +663,34 @@ private[spark] class BlockManager(
     require(blockId != null, "BlockId is null")
     var runningFailureCount = 0
     var totalFailureCount = 0
-    val locations = getLocations(blockId)
+
+    // Because all the remote blocks are registered in driver, so it is not necessary to ask
+    // all the slave executors to get block status.
+    val locationAndStatus = master.getLocationsAndStatus(blockId)
+
+    val blockSize = locationAndStatus._2.map { status =>
+      // Disk size and mem size cannot co-exist, so it's ok to sum them together to get block size.
+      status.diskSize + status.memSize
```

Comment:
I think it's dangerous, because you assume the default value of diskSize/memSize is always 0, but this is not guaranteed.

Reply:
@jiangxb1987 would you please explain more? I'm not quite following your comment. Are you referring to the below line `  }.getOrElse(0L)`?

Reply:
Are you saying we need to check `StorageLevel` to decide whether to use diskSize or memSize as block size?

Reply:
The default value of diskSize or memSize can be 0 or -1 or even other numbers, it's not guaranteed. Maybe we can use `Math.max(status.diskSize, status.memSize)` instead?

Reply:
I get your point, also thinking of using `Math.max` instead.